extensions/nginx/acme.js

```diff
@@ -0,0 +1,91 @@
+}
+
+function install(ui, task) {
+    if (isInstalled()) {
+        task.skip('acme.sh is already installed');
+    }
+
+    const acmeTmpDir = path.join(os.tmpdir(), 'acme.sh');
+    const acmeApiUrl = 'https://api.github.com/repos/Neilpang/acme.sh/releases/latest';
+
+    ui.logVerbose('ssl: creating /etc/letsencrypt directory', 'green');
+
+    // acme.sh creates the directory without global read permissions, so we need to make
+    // sure it has global read permissions first
+    return ui.sudo('mkdir -p /etc/letsencrypt').then(() => {
+        ui.logVerbose('ssl: downloading acme.sh to temporary directory', 'green');
+        return fs.emptyDir(acmeTmpDir)
+    }).then(() => got(acmeApiUrl)).then((response) => {
+        if (response.statusCode !== 200) {
+            return Promise.reject(new cli.errors.CliError('Unable to query GitHub for ACME download URL'));
+        }
+
+        try {
+            response = JSON.parse(response.body).tarball_url;
+        } catch (e) {
+            return Promise.reject(new cli.errors.CliError('Unable to parse Github api response for acme'));
+        }
+
+        return download(response, acmeTmpDir, {extract: true});
+    }).then(() => {
+        // The archive contains a single folder with the structure
+        //  `{user}-{repo}-{commit}`, but we don't know what commit is
+        //  from the API call. Since the dir is empty (we cleared it),
+        //  the only thing in acmeTmpDir will be the extracted zip.
+        const acmeCodeDir = path.resolve(acmeTmpDir, fs.readdirSync(acmeTmpDir)[0]);
+
+        ui.logVerbose('ssl: installing acme.sh components', 'green');
+
+        // Installs acme.sh into /etc/letsencrypt
+        return ui.sudo('./acme.sh --install --home /etc/letsencrypt', {cwd: acmeCodeDir});
+    }).catch((error) => Promise.reject(new cli.errors.ProcessError(error)));
+}
+
+function generateCert(ui, domain, webroot, email, staging) {
+    const cmd = `/etc/letsencrypt/acme.sh --issue --home /etc/letsencrypt --domain ${domain} --webroot ${webroot} ` +
+    `--reloadcmd "nginx -s reload" --accountemail ${email}${staging ? ' --staging' : ''}`;
+
+    return ui.sudo(cmd).catch((error) => {
+        if (error.code === 2) {
+            // error code 2 is given if a cert doesn't need to be renewed
+            return Promise.resolve();
+        }
+
+        if (error.stderr.match(/Verify error:(Fetching|Invalid Response)/)) {
+            // Domain verification failed
+            return Promise.reject(new cli.errors.SystemError(
+                'Your domain name is not pointing to the correct IP address of your server, please update it and run `ghost setup ssl` again'
+            ));
+        }
+
+        // It's not an error we expect might happen, throw a ProcessError instead.
+        return Promise.reject(new cli.errors.ProcessError(error));
+    });
+}
+
+function remove(domain, ui) {
+    const acmeHome = path.join(os.homedir(), '.acme.sh');
```

Comment:
Isn't the ACME install location `/etc/letsencrypt`?

Reply:
not in this case, no. We're migrating from the original acme install location to the new location, which means we want to use the old location here.

I can update the variable name to be more clear though

Reply:
Got it. Should the nginx uninstall hook be calling something similar to this to delete the certificate from the system? This might be a separate issue 🤔

Reply:
hmm yeah that's not a bad idea